demo/admin/src/products/highlights/SelectProductFieldGrid.tsx

```diff
@@ -0,0 +1,106 @@
+import { gql, useQuery } from "@apollo/client";
+import {
+    type GridColDef,
+    muiGridFilterToGql,
+    muiGridSortToGql,
+    useBufferedRowCount,
+    useDataGridRemote,
+    usePersistentColumnState,
+} from "@comet/admin";
+import { DataGridPro, type GridRowSelectionModel } from "@mui/x-data-grid-pro";
+import { useState } from "react";
+
+import {
+    type GQLSelectProductFieldGridFragment,
+    type GQLSelectProductFieldGridQuery,
+    type GQLSelectProductFieldGridQueryVariables,
+} from "./SelectProductFieldGrid.generated";
+
+type Props = {
+    onSelect?: (product: GQLSelectProductFieldGridFragment | null) => void;
+};
+export function SelectProductFieldGrid(props: Props) {
```

Comment:
The Admin Generator already supports this form of select grid, should we generate it?

https://github.com/vivid-planet/comet/pull/2449

(Less handmade code to maintain)

Reply:
yes, but I didn't want to mix handmade with generated